src/Service/Compare.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\Smartphone;
+
+/**
+ * Compare class compares two smartphones.
+ */
+class Compare {
+
+  /**
+   * Takes two smartphones and compares them based to their specifications and
+   * price and returns results as an associative array.
+   *
+   * @param \App\Entity\Smartphone $smartphone1
+   *   First smartphone object.
+   * @param \App\Entity\Smartphone $smartphone2
+   *   Second smartphone object.
+   *
+   * @return array
+   *   Returns comparison result as an associative array.
+   */
+  public static function compare(Smartphone $smartphone1, Smartphone $smartphone2):array {
+    $result = [];
+    $result[$smartphone1->getName()] = 0;
+    $result[$smartphone2->getName()] = 0;
+
+    if ($smartphone1->getFrontCamera() >
+      $smartphone2->getFrontCamera()) {
+      $result['front_camera'] = $smartphone1->getName();
+      $result[$smartphone1->getName()]++;
+    }
+    elseif ($smartphone1->getFrontCamera() <
+      $smartphone2->getFrontCamera()) {
+      $result['front_camera'] = $smartphone2->getName();
+      $result[$smartphone2->getName()]++;
+    }else{
+      $result['front_camera'] = 'Same';
+    }
+
+    if ($smartphone1->getBackCamera() >
```

Comment:
Create e helper function where comparsion is done and we will pass the items we want to compare as parameter.

It can be like this

```
   function compareSpecs($spec1, $spec2, $name1, $name2, $specName): string {
    if ($spec1 > $spec2) {
      return $name1;
    } elseif ($spec1 < $spec2) {
      return $name2;
    } else {
      return "Same";
    }
  }
```
  
`$result['front_camera'] = self::compareSpecs($smartphone1->getFrontCamera(), $smartphone2->getFrontCamera(), $smartphone1->getName(), $smartphone2->getName(), 'front camera');`